File: src/numericalrecipes/Ch6_SpecialFunctions/gammaln/speed_gammaln.f90
program main
   ! COMPILATION:
   ! gfortran -c gammaln.f90 speed_gammaln.f90
   ! gfortran gammaln.o speed_gammaln.o -o fspeed_gammaln
   use mod_gammaln
   implicit none
   real(8)                 :: x, gl, time_exec
   integer                 :: i, c1, c2, cr

   ! Input data
   x = 0.25

   ! Timing of the for loop
   call system_clock(count_rate=cr)
   call system_clock(c1)

   do i = 1, 1000000
      gl = gammaln(x)
   end do

   call system_clock(c2)

   time_exec = (c2 - c1) / real(cr)

   write(*,*) gl, time_exec
end program main

File: src/numericalrecipes/Ch6_SpecialFunctions/gammaln/speed_gammaln.cpp
// COMPILATION:
// g++ gammaln.cpp speed_gammaln.cpp -o cppSpeed_gammaln
#include <iostream>
#include <chrono>
using namespace std;

double gammaln(float x);


int main() {
   /* Function that measures elapsed time of a for loop
      that calculates gammaln */

   // Input data
   float gl, x = 0.25;

   // Timing of the for loop
   auto start = chrono::steady_clock::now();

   for(int i=0; i<1e6; i++){
      gl = gammaln(x);
   }

   auto end = chrono::steady_clock::now();

   cout << gl << " " << chrono::duration_cast<chrono::milliseconds>(end - start).count()
      << endl;
   return 0;
}
